llvm/tools/llvm-objdump/llvm-objdump.cpp

```diff
@@ -693,6 +693,32 @@ class PrettyPrinter {
     } else
       OS << "\t<unknown>";
   }
+
+  virtual std::string getInstructionSeparator() const { return "\n"; }
+
+  virtual void emitPostInstructionInfo(formatted_raw_ostream &FOS,
+                                       const MCAsmInfo &MAI,
+                                       const MCSubtargetInfo &STI,
+                                       StringRef Comments,
+                                       LiveVariablePrinter &LVP) {
+    do {
+      if (!Comments.empty()) {
```

Comment:
If Comments is a multi-line comment, `getInstructionSeparator();` will be printed multiple times, which is not correct. 

This is a case where we should probably just duplicate emitPostInstructionInfo for Hexagon and not introduce a new virtual function `getInstructionSeparator`